onnx/defs/tensor/defs.cc

```diff
@@ -136,7 +171,70 @@ ONNX_OPERATOR_SCHEMA(Concat)
     .TypeConstraint(
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
-        "Constrain output types to float tensors.");
+        "Constrain output types to float tensors.")
+    .ShapeInferenceFunction([](InferenceContext& ctx) {
+        if (!hasExactlyNInputTypes(ctx, 1, "Concat")) {
+          return;
+        }
+
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+        if (ctx.getNumInputs() == 0) {
```

Comment:
I don't understand: there's a check earlier for exactly 1 input type.

Reply:
++